core/config/coreMintKeeper.js

```diff
@@ -0,0 +1,43 @@
+import { makePrivateName } from '../../util/PrivateName';
+
+export function makeCoreMintKeeper() {
+  // An asset can either be a purse or payment. An asset keeper
+  // keeps track of either all of the purses (purseKeeper) or all
+  // of the payments (paymentKeeper) and their respective amounts.
+  function makeAssetKeeper(type) {
+    // asset to amount
+    const assets = makePrivateName();
+    return {
+      updateAmount(asset, newAmount) {
+        assets.set(asset, newAmount);
+      },
+      recordNew(asset, initialAmount) {
+        assets.init(asset, initialAmount);
+      },
+      getAmount(asset) {
+        return assets.get(asset);
+      },
+      getType() {
+        return `${type}`;
```

Comment:
Please don't call these types. Let's postpone the term "type" until we have more of a type system, via idl. Types will need identity. The most you could say about this method is it returns an unqualified type name, and is thereby subject to collisions. I think you want a different concept. Like purse names, these should be seen primarily as an aid to debugging rather than an inherent part of the meaning of the program.

Reply:
Hi Mark, my apologies. I know you had mentioned this exact change before, and I took it out then, but it appears to have crept back due to all of the merging (from master, from other PRs) that has happened on this branch. I'll retract my request for review and make sure that nothing like this remains.

Reply:
Apologies unnecessary, but thanks for the explanation.

As long as your retracting, revising, and resubmitting, could you put in necessary `harden` calls as well? Thanks.

Reply:
Yes, of course